src/errors.ts

```diff
@@ -0,0 +1,343 @@
+import cwdRelativePath from "./cwd-relative-path";
+import {ErrorCode} from "./error-codes";
+import normalizeSeparators from "./normalize-separators";
+import rootRelativePath from "./root-relative-path";
+import {ErrorDetails, Options} from "./types";
+
+/**
+ * Create an error indicating that the file could not be parsed.
+ *
+ * @param file The path of the file that could not be parsed.
+ * @param detail The reason why the file could not be parsed.
+ * @returns The error details.
+ */
+export const couldNotParse = (file: string, detail: string): ErrorDetails => ({
+    markerID: null,
+    code: ErrorCode.couldNotParse,
+    reason: `Could not parse ${file}: ${detail}`,
+});
+
+/**
+ * Create an error indicating that a different comment style is being used.
+ *
+ * @param markerID The ID of the marker that is affected.
+ * @param line The line number where the marker is located.
+ * @returns The error details.
+ */
+export const differentCommentSyntax = (markerID: string, line: number) => ({
+    markerID,
+    reason: `Sync-start tags for '${markerID}' given in different comment styles. Please use the same style for all sync-start tags that have identical identifiers.`,
+    location: {line},
+    code: ErrorCode.differentCommentSyntax,
+});
+
+/**
+ * Create an error indicating a duplicate marker declaration.
+ *
+ * @param markerID The ID of the marker that is duplicated.
+ * @param line The line number where the marker is located.
+ * @returns The error details.
+ */
+export const duplicateMarker = (
+    markerID: string,
+    line: number,
+): ErrorDetails => ({
+    markerID,
+    reason: `Sync-tag '${markerID}' declared multiple times`,
+    location: {line},
```

Comment:
same ide ahere

Reply:
As before, checksync only reports errors for each tag location. The onus is currently on the user to go open the file and investigate. Perhaps in a future update I can rework things to provide even more context, but at the moment we don't report multiple locations for one error.